NetTopologySuite.IO.PostGis/PostGisWriter.cs

```diff
@@ -170,20 +149,20 @@ private static void Write(ICoordinateSequence sequence, Ordinates ordinates, Bin
                 if ((ordinates & Ordinates.Z) != 0)
                 {
                     double z = sequence.GetOrdinate(i, Ordinate.Z);
-                    if (double.IsNaN(z)) z = 0d;
+                    Debug.Assert(!double.IsNaN(z));
                     writer.Write(z);
                 }
                 if ((ordinates & Ordinates.M) != 0)
                 {
                     double m = sequence.GetOrdinate(i, Ordinate.M);
-                    if (double.IsNaN(m)) m = 0d;
+                    Debug.Assert(!double.IsNaN(m));
```

Comment:
We should just skip these two `Debug.Assert` lines (this one, for M, and the other one, for Z).  Nobody's ever going to see it since we don't release debug builds, and it can trigger even if the rest of our code is correct (since it depends fully on the input data).

This is assuming that I'm not wrong in my understanding that `Debug.Assert` calls are meant to inform the developer that, if the rest of our code is correct, its condition will always be true at the point in time when the `Debug.Assert` call is made.

Reply:
If I'm wrong and these `Debug.Assert` lines do in fact fulfill a purpose, then please use an overload such as [`Debug.Assert(bool, string)`](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.debug.assert?view=netframework-4.7.2#System_Diagnostics_Debug_Assert_System_Boolean_System_String_) that allows you to provide a message explaining in more detail what happened and, perhaps, why it's bad.

Reply:
Yeah, this is mainly meant for failing tests, to alert us earlier and directly as to the broken assumption. I added an informative message and I think it has some value, but if you really feel that it's unnecessary let me know and I'll remove it.

Reply:
> if you really feel that it's unnecessary let me know and I'll remove it.
Per [@peetw's testing](https://github.com/NetTopologySuite/NetTopologySuite.IO.PostGis/pull/9#discussion_r248848731), I think this (and its counterpart assert for Z) should just go away.  It sounds like we'd faithfully preserve whatever values are carried by the instance, in which case we don't really have much of a reason to fret about it at this layer of the system.

Reply:
Agreed.